src/libopensc/card-npa.c

```diff
@@ -415,6 +399,18 @@ static int npa_init(sc_card_t * card)
 	if (r != SC_SUCCESS)
 		goto err;
 
+	size_t bufsize = sizeof dir_content_ref;
+	u8 *p = dir_content_ref;
+	while (bufsize > 0) {
+		if (card->app_count == SC_MAX_CARD_APPS) {
```

Comment:
is this branch reachable when you removed the `sc_enum_apps()`? Or is this called from some general match/detect/init code?

Reply:
indeed.

but I think I could do better here. Since after this change, the reference data is hard coded, I could also hardcode the initialized card apps without parsing the buffer. I'l look into this later...

Reply:
done